testsuite/mockserver.py

```diff
@@ -4,30 +4,60 @@
 import httpx
 from apyproxy import ApyProxy
 
+from testsuite.openshift.client import OpenShiftClient
 from testsuite.utils import ContentType
+from testsuite.gateway import Referencable
 
 
-class Mockserver:
+class Mockserver(Referencable):
     """
     Mockserver deployed in Openshift (located in Tools or self-managed)
     """
 
-    def __init__(self, url):
+    def __init__(self, url, service_name: str, tools: OpenShiftClient):
         self.client = ApyProxy(url, session=httpx.Client(verify=False, timeout=5))
+        self.service_name = service_name
+        self.tools = tools
 
-    def _expectation(self, expectation_id, response_data):
+    @property
+    def reference(self):
```

Comment:
this does not feel right, as the mockserver itself is just a client to a any Mockserver even deployed without openshift.

Reply:
I would suggest creating another Backend implementation for mockserver and replacing backend fixture

Reply:
> this does not feel right, as the mockserver itself is just a client to a any Mockserver even deployed without openshift.

but our tests will always use mockserver deployed inside the k8s as part of the tools, am I missing something?

> I would suggest creating another Backend implementation for mockserver and replacing backend fixture

I think another mockserver implementation is unnecessary here, it creates additional overhead and everything backend mockserver require is a reference to his service. And backend fixture has already been replaced

Reply:
> but our tests will always use mockserver deployed inside the k8s as part of the tools, am I missing something?

Tools can be used from a different cluster or you can just use external mockserver as well

> I think another mockserver implementation is unnecessary here, it creates additional overhead and everything backend mockserver require is a reference to his service. And backend fixture has already been replaced

You would either deploy httpbin or mockserver, I do not see a problem, besides if we agree that mockserver always need to be on the same cluster than you do not have to deploy it.